server/app/views/applicant/IneligibleBlockView.java

```diff
@@ -53,11 +61,25 @@ public Content render(
       ReadOnlyApplicantProgramService roApplicantProgramService,
       Messages messages,
       long applicantId,
-      ProgramDefinition programDefinition) {
+      ProgramDefinition programDefinition,
+      Optional<String> blockId)
+      throws ProgramBlockDefinitionNotFoundException {
     long programId = roApplicantProgramService.getProgramId();
     boolean isTrustedIntermediary = submittingProfile.isTrustedIntermediary();
     String programDetailsLink = programDefinition.externalLink();
     ATag infoLink = null;
+    String eligibilityMsg = "";
+    boolean isEligibilityMsgEnabled =
+        settingsManifest.getCustomizedEligibilityMessageEnabled(request);
+    if (blockId.isPresent()) {
+      BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockId.get());
+      Locale preferredLocale = messages.lang().toLocale();
+      eligibilityMsg =
+          blockDefinition
+              .localizedEligibilityMessage()
+              .map(localizedStrings -> localizedStrings.maybeGet(preferredLocale).orElse(""))
```

Comment:
**suggestion:** Could you do localizedStrings.getOrDefault(preferredLocale)?

Reply:
yeah! changed it to getOrDefault()